programming/sr/cheat_sheet.md

```diff
@@ -0,0 +1,264 @@
+
+# Code here runs before the start button is pressed
+
+R.wait_start()
+~~~~~
+
+### Initialisation with extra logging
+
+You can also tell the robot to print extra logging information, although this is quite noisy.
+
+~~~~~ python
+R = Robot(verbose=True)
+~~~~~
+
+## Selecting which board to control
+
+If you only have one board of a given type plugged into your robot, then you can use its singular name:
+
+~~~~~ python
+R.power_board.something
+R.motor_board.something
+R.servo_board.something
+R.ruggeduino.something
+~~~~~
+
+If you have multiple boards of a given type plugged into your robot, you must index them by serial number:
+
+~~~~~ python
+R.motor_boards["srABC1"].something
+R.servo_boards["srXYZ2"].something
+R.ruggeduinos["1234567890"].something
+~~~~~
+
+## Power Board
+
+The outputs on the power board will turn on when you initialise your robot.
+
+### Turn on and off the power outputs
+
+~~~~~ python
+# Turn all of the outputs on
+R.power_board.outputs.power_on()
+
+# Turn all of the outputs off
+R.power_board.outputs.power_off()
+
+# Turn a single output on
+R.power_board.outputs[OUT_H0].is_enabled = True
+
+# Turn a single output off
+R.power_board.outputs[OUT_H0].is_enabled = False
+~~~~~
+
+### Reading voltage and current
+
+~~~~~ python
+# Read the current of an individual output
+current = R.power_board.outputs[OUT_H0].current
+
+# Read the current and voltage from the LiPo battery
+voltage = R.power_board.battery_sensor.voltage
+current = R.power_board.battery_sensor.current
+~~~~~
+
+### Buzzer
+
+The power board has an on-board piezoelectric buzzer.
+
+~~~~~ python
+# Play a standard note C6 -> C8 included for 0.5s
+R.power_board.piezo.buzz(0.5, Note.C6)
+
+# Play a tone at 1047Hz for 1 second
+R.power_board.piezo.buzz(1, 1047)
+~~~~~
+
+## Motors
+
+### Powering Motors
+
+You can set the power of each motor on the board between -1 and 1.
+
+If you change the power of your motor too rapidly, the overcurrent protection may be triggered.
+
+~~~~~ python
+R.motor_board.motors[0].power = 1
+R.motor_board.motors[1].power = -1
+~~~~~
+
+Setting a motor to `COAST` is equivalent to power level `0`.
+
+~~~~~ python
+# This is the same operation
+R.motor_board.motors[0].power = COAST
+R.motor_board.motors[0].power = 0
+~~~~~
+
+### Braking Motors
+
+You can also brake a motor, which will quickly slow the motor.
+
+~~~~~ python
+R.motor_board.motors[0].power = BRAKE
+R.motor_board.motors[1].power = -1
+~~~~~
+
+## Servos
+
+You can set the position of each servo output on the board between -1 and 1.
+
+~~~~~ python
+R.servo_board.servos[0].position = -1
+R.servo_board.servos[1].position = 1
+~~~~~
+
+You can also set the position to `0`, which is the approximate centre.
+
+This is different to setting the position to `None`, which will unpower the servo.
+
+~~~~~ python
+# This servo is now unpowered, and will move more freely.
+R.servo_board.servos[11].position = None
+~~~~~
+
+## Camera
+
+## Taking a photo
```

Comment:
Should this (and "Looking for markers") be level-3 headings rather than level-2?
```suggestion
### Taking a photo
```